homeassistant/components/esphome/__init__.py

```diff
@@ -352,6 +353,8 @@ def _async_setup_device_registry(
     configuration_url = None
     if device_info.webserver_port > 0:
         configuration_url = f"http://{entry.data['host']}:{device_info.webserver_port}"
+    elif dashboard := async_get_dashboard(hass):
+        configuration_url = f"homeassistant://hassio/ingress/{dashboard.addon_slug}"
```

Comment:
Maybe the dashboard module should keep the add-on details and just expose a config url?

Reply:
Well the challenge here that `homeassistant://` is a special URL prefix that works for `configuration_url`, but not for other locations.

Reply:
I also don't think that we will need it anywhere else.

Reply:
Makes sense. 👍